### PR TITLE
Mongo db configuration

### PR DESCRIPTION
**Pull request by kudlaty9102**

### CODE REVIEW

File: src/main/java/org/grizz/model/Building.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.Builder;
+import lombok.Data;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+@Data
+@Builder
+@Document(collection = "buildings")
+public class Building {
+
+    private String type;
```
Comment (tomekbielaszewski) [inline src/main/java/org/grizz/model/Building.java]: How about changing this with enum type? `BuildingType.SOLAR_POWER_PLANT`, `BuildingType.METAL_MINE`, `BuildingType.CRYSTAL_MINE` etc

File: src/test/java/org/grizz/web/api/SampleData.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = OgameCloneApplication.class)
+@WebAppConfiguration
+public class SampleData {
+
+    @Autowired
+    PlanetRepository planetRepository;
+
+    // @Ignore
```
Comment (tomekbielaszewski) [inline src/test/java/org/grizz/web/api/SampleData.java]: Uncomment

File: src/main/java/org/grizz/model/Building.java (hunk 1 above)
Comment (kudlaty9102) [inline src/main/java/org/grizz/model/Building.java]: Hmmm... I forgot about that. I propose create new Issues wherein we list the all types of buildings.

Comment (tomekbielaszewski) [inline src/main/java/org/grizz/model/Building.java]: Sure, lets move it to another Issue then. I will take care of that :)

Comment (tomekbielaszewski) [inline src/main/java/org/grizz/model/Building.java]: Here you go #33

File: src/test/java/org/grizz/web/api/SampleData.java (hunk 2 above)
Comment (tomekbielaszewski) [inline src/test/java/org/grizz/web/api/SampleData.java]: CRITICAL: Move this test to another package. It has nothing in common with web.api!!

File: src/main/java/org/grizz/model/Building.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package org.grizz.model;
+
+import lombok.Builder;
+import lombok.Data;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+@Data
+@Builder
+@Document(collection = "buildings")
```
Comment (tomekbielaszewski) [inline src/main/java/org/grizz/model/Building.java]: Buildings should not be stored in mongo on it's own. Building cannot exist without planet so lets keep all buildings inside plannets collection as a part of planet.

File: src/main/java/org/grizz/model/Planet.java [hunk 4]
```diff
@@ -1,10 +1,19 @@
...
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.util.*;
+
 @Data
 @Builder
+@Document(collection = "planets")
 public class Planet {
+
     private String id;
```
Comment (tomekbielaszewski) [inline src/main/java/org/grizz/model/Planet.java]: Annotate id with @Id

File: src/main/java/org/grizz/model/User.java [hunk 5]
```diff
@@ -2,13 +2,17 @@
...
 import lombok.Data;
+import org.springframework.data.mongodb.core.mapping.Document;
 
 import java.util.Set;
 
 @Data
 @Builder
+@Document(collection = "users")
 public class User {
+
```
Comment (tomekbielaszewski) [inline src/main/java/org/grizz/model/User.java]: Add String id and annotate it with @Id

File: src/main/java/org/grizz/model/Planet.java (hunk 4 above)
Comment (kudlaty9102) [inline src/main/java/org/grizz/model/Planet.java]: Maybe we should add separated fields: id and planetName? Id will be as a db id and planetName will be from player? In Controller we add method findByName.

Comment (tomekbielaszewski) [inline src/main/java/org/grizz/model/Planet.java]: Are you assuming that planetNames will be unique? IMHO planets should be fetched by their coordinates in galaxy at most. I even do find useless fetching planets by ID in this case. But leave it as is for now.
And sure - we need planetName also, but not as an unique field

**Pull request merged**